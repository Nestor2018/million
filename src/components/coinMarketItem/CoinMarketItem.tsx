import React, { memo } from 'react';
import { View, Text, StyleSheet } from 'react-native';

import colors from '../../utils/colors';

interface CoinMarketItemProps {
  market: {
    name: string;
    price_usd: string;
  };
}

/**
 * Component CoinMarketItem (Renders the coin market)
 * @param {CoinMarketItemProps} props - Props containing the market object
 * @returns {JSX.Element} The rendered component
 */
const CoinMarketItem: React.FC<CoinMarketItemProps> = memo(({ market }) => {
  return (
    <View style={styles.container}>
      <Text
        style={styles.nameText}
        testID="marketName"
        accessible={true}
        accessibilityLabel={`${market.name}`}
      >
        {market.name}
      </Text>
      <Text
        style={styles.priceText}
        testID="marketPrice"
        accessible={true}
        accessibilityLabel={`${market.price_usd}`}
      >
        {market.price_usd}
      </Text>
    </View>
  );
});

const styles = StyleSheet.create({
  container: {
    backgroundColor: 'rgba(0,0,0, 0.1)',
    borderColor: colors.zircon,
    borderWidth: 1,
    padding: 16,
    margin: 8,
    alignItems: 'center',
  },
  nameText: {
    color: colors.white,
    fontWeight: 'bold',
  },
  priceText: {
    color: colors.white,
  },
});

export default CoinMarketItem;
