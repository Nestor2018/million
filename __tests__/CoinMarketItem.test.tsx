import React from 'react';
import { render } from '@testing-library/react-native';
import CoinMarketItem from '../src/components/coinMarketItem/CoinMarketItem';

describe('CoinMarketItem component', () => {
  const mockMarket = {
    name: 'Market Name',
    price_usd: '100',
  };

  it('renders correctly', () => {
    const { getByText } = render(<CoinMarketItem market={mockMarket} />);
    const nameElement = getByText(mockMarket.name);
    const priceElement = getByText(mockMarket.price_usd);
    expect(nameElement).toBeTruthy();
    expect(priceElement).toBeTruthy();
  });

  it('is accessible with proper accessibility labels', () => {
    const { getByTestId } = render(<CoinMarketItem market={mockMarket} />);

    const nameElement = getByTestId('marketName');
    expect(nameElement.props.accessible).toBe(true);
    expect(nameElement.props.accessibilityLabel).toBe('Market Name');

    const priceElement = getByTestId('marketPrice');
    expect(priceElement.props.accessible).toBe(true);
    expect(priceElement.props.accessibilityLabel).toBe('100');
  });

  it('does not re-render when props do not change', () => {
    const { rerender, getByTestId } = render(<CoinMarketItem market={mockMarket} />);

    const initialRender = getByTestId('marketName');
    expect(initialRender).toBeTruthy();

    // Re-renderiza el componente con las mismas props
    rerender(<CoinMarketItem market={mockMarket} />);
    const afterRerender = getByTestId('marketName');

    // Verifica que el componente no haya cambiado
    expect(afterRerender).toBe(initialRender);
  });

  it('matches snapshot', () => {
    const tree = render(<CoinMarketItem market={mockMarket} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
