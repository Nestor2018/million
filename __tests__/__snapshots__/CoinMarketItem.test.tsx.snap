// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoinMarketItem component matches snapshot 1`] = `
<View>
  <Text
    accessibilityLabel="Market Name"
    accessible={true}
    testID="marketName"
  >
    Market Name
  </Text>
  <Text
    accessibilityLabel="100"
    accessible={true}
    testID="marketPrice"
  >
    100
  </Text>
</View>
`;
